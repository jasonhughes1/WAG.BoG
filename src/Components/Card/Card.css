.card {
  border: 1px solid black;
  margin: 20px;
  background: lightgrey;
  height: 430px;
  width: 275px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  border-radius: 25px;
  box-shadow: 2px 4px 4px 2px; }

.street {
  margin: -20px; }

.state {
  margin: -20px;
  padding-bottom: 30px; }

.sex {
  margin-top: -15px; }

.card:hover {
  cursor: pointer;
  background-color: white;
  border: 1px solid white; }

.zip {
  margin-top: -15px; }

.picture {
  height: 200px;
  width: 200px;
  border: 10px solid black; }

.inactive {
  cursor: pointer;
  font-weight: bold;
  color: red;
  margin-top: 15px; }
