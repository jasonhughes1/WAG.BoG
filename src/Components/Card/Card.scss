.card {
  border: 1px solid black;
  margin: 20px;
  background: lightgrey;
  height: 430px;
  width: 275px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  border-radius: 25px;
  box-shadow: 2px 4px 4px 2px;
}


.street {
  margin: -20px;
}

.state {
  margin: -20px;
  padding-bottom: 30px;
}

.sex {
  margin-top: -15px;
}

.card:hover {
  cursor: pointer;
  background-color: white;
  border: 1px solid white;
}

.zip {
  margin-top: -15px;
}

.picture {
  height: 200px;
  width: 200px;
  border: 10px solid black;
}

.favorite-button {
  padding-left: 10px;
  background-color: white;
  border: 1px solid black;
  height: 100px;
  width: 60px;
  cursor: pointer;
  font-weight: bold;
  color: blue;
  margin-top: 15px;
  box-shadow: 0px 0px 10px;
}

.active {
  color: red;
}
