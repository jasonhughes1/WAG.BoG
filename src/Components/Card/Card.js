import React from 'react';
import './Card.css';
import { connect } from 'react-redux';
import { addFavorites, removeFavorites } from '../../Actions/Actions';
import PropTypes from 'prop-types';


export class Card extends React.Component  {
  constructor(props) {
    super(props);
    this.state = {active: false};
  }

  filterFavs = (favIndex) => {
    let results = this.props.favorites.filter(fav => fav.name === favIndex);
    return results;
  };

  toggleFavorites = (dogCard) => {
    this.setState({active: true});
    if (!this.filterFavs(dogCard.name).length) {
      return this.props.addFavorites(dogCard);
    } else {
      this.setState({active: false});
      return this.props.clearFavorites(dogCard);
    }
  };
  render() {
    return (
      <div className='card'>
        <span className={this.state.active === false ? 'favorite-button' : 'favorite-button active'} onClick={() => this.toggleFavorites(this.props)}>Favorite</span>
        <h2 className='name'> {this.props.name}</h2>
        <img className ='picture' src={this.props.picture} />
        <h3 className='breed'>Breed:  {this.props.breed}</h3>
        <h4 className='sex'>Sex: {this.props.sex}</h4>
        <p className='street'>{this.props.street}</p>
        <p className='city'>{this.props.city}</p>
        <p className='state'>{this.props.state}</p>
        <p className='zip'>{this.props.zip}</p>
      </div>
    );
  }
}
export const mapStateToProps = (store) => {
  return {
    favorites: store.favorites
  };
};

export const mapDispatchToProps = dispatch => ({
  addFavorites: (favorites) => {
    dispatch(addFavorites(favorites));
  },
  clearFavorites: (favorites) => {
    dispatch(removeFavorites(favorites));
  }
});

Card.propTypes = {
  addFavorites: PropTypes.func,
  breed: PropTypes.string,
  city: PropTypes.string,
  clearFavorites: PropTypes.func,
  favorites: PropTypes.array,
  name: PropTypes.string,
  picture: PropTypes.string,
  sex: PropTypes.string,
  state: PropTypes.string,
  zip: PropTypes.string,
  street: PropTypes.string
};

export default connect(mapStateToProps, mapDispatchToProps)(Card);
