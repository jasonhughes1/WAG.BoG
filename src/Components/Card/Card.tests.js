import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card';

describe('Card tests', () => {
  let card;

  beforeEach( () => {
    card = shallow(<Card />);
  });

  it('should be defined', () => {
    expect(card).toBeDefined();
  });

  it('should match the snapshot', () => {
    expect(card).toMatchSnapshot();
  });
});
