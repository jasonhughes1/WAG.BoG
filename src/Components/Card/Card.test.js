import React from 'react';
import { shallow } from 'enzyme';
import { Card, mapStateToProps, mapDispatchToProps} from './Card';

describe('Card tests', () => {
  let card;

  beforeEach( () => {
    const mockFunc = jest.fn();
    card = shallow(<Card
      addFavorites={mockFunc}
      breed={'lab'}
      city= {'denver'}
      clearFavorites={mockFunc}
      favorites={['favs']}
      name={'brooklyn'}
      picture={'dogfinder.jpg'}
      sex={'M'}
      state={'Colorado'}
      zip={'80223'}
      street={'colfax'}
    />);
  });

  it('should be defined', () => {
    expect(card).toBeDefined();
  });

  it('should match the snapshot', () => {
    expect(card).toMatchSnapshot();
  });
});

describe('map state to props', () => {
  it('should receive favorites from the store', () => {
    const mockStore = {
      favorites: []
    };
    const expected = mapStateToProps(mockStore);
    expect(expected.favorites).toEqual(mockStore.favorites);
  });
});

describe('mapDispatchToProps', () => {
  let mockDispatch;
  let result;

  beforeEach(() => {
    mockDispatch = jest.fn();
    result = mapDispatchToProps(mockDispatch);
  });

  it('should call dispatch when addFavorites is called', () => {
    result.addFavorites();
    expect(mockDispatch).toHaveBeenCalled();
  });

  it('should call dispatch when clearFavorites is called', () => {
    result.clearFavorites();
    expect(mockDispatch).toHaveBeenCalled();
  });
});
