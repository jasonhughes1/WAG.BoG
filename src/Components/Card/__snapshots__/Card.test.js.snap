// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card tests should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    addFavorites={[Function]}
    breed="lab"
    city="denver"
    clearFavorites={[Function]}
    favorites={
        Array [
            "favs",
          ]
    }
    name="brooklyn"
    picture="dogfinder.jpg"
    sex="M"
    state="Colorado"
    street="colfax"
    zip="80223"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="favorite-button"
          onClick={[Function]}
>
          Favorite
</span>,
        <h2
          className="name"
>
           
          brooklyn
</h2>,
        <img
          className="picture"
          src="dogfinder.jpg"
/>,
        <h3
          className="breed"
>
          Breed:  
          lab
</h3>,
        <h4
          className="sex"
>
          Sex: 
          M
</h4>,
        <p
          className="street"
>
          colfax
</p>,
        <p
          className="city"
>
          denver
</p>,
        <p
          className="state"
>
          Colorado
</p>,
        <p
          className="zip"
>
          80223
</p>,
      ],
      "className": "card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Favorite",
          "className": "favorite-button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Favorite",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            " ",
            "brooklyn",
          ],
          "className": "name",
        },
        "ref": null,
        "rendered": Array [
          " ",
          "brooklyn",
        ],
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "picture",
          "src": "dogfinder.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Breed:  ",
            "lab",
          ],
          "className": "breed",
        },
        "ref": null,
        "rendered": Array [
          "Breed:  ",
          "lab",
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Sex: ",
            "M",
          ],
          "className": "sex",
        },
        "ref": null,
        "rendered": Array [
          "Sex: ",
          "M",
        ],
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "colfax",
          "className": "street",
        },
        "ref": null,
        "rendered": "colfax",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "denver",
          "className": "city",
        },
        "ref": null,
        "rendered": "denver",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Colorado",
          "className": "state",
        },
        "ref": null,
        "rendered": "Colorado",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "80223",
          "className": "zip",
        },
        "ref": null,
        "rendered": "80223",
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="favorite-button"
            onClick={[Function]}
>
            Favorite
</span>,
          <h2
            className="name"
>
             
            brooklyn
</h2>,
          <img
            className="picture"
            src="dogfinder.jpg"
/>,
          <h3
            className="breed"
>
            Breed:  
            lab
</h3>,
          <h4
            className="sex"
>
            Sex: 
            M
</h4>,
          <p
            className="street"
>
            colfax
</p>,
          <p
            className="city"
>
            denver
</p>,
          <p
            className="state"
>
            Colorado
</p>,
          <p
            className="zip"
>
            80223
</p>,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Favorite",
            "className": "favorite-button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Favorite",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " ",
              "brooklyn",
            ],
            "className": "name",
          },
          "ref": null,
          "rendered": Array [
            " ",
            "brooklyn",
          ],
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "picture",
            "src": "dogfinder.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Breed:  ",
              "lab",
            ],
            "className": "breed",
          },
          "ref": null,
          "rendered": Array [
            "Breed:  ",
            "lab",
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Sex: ",
              "M",
            ],
            "className": "sex",
          },
          "ref": null,
          "rendered": Array [
            "Sex: ",
            "M",
          ],
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "colfax",
            "className": "street",
          },
          "ref": null,
          "rendered": "colfax",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "denver",
            "className": "city",
          },
          "ref": null,
          "rendered": "denver",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Colorado",
            "className": "state",
          },
          "ref": null,
          "rendered": "Colorado",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "80223",
            "className": "zip",
          },
          "ref": null,
          "rendered": "80223",
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
